### ISSUE TITLE
Agent-install.sh no longer successfully processes an agent device upgrade

### ISSUE THREAD

**Issue by omega2733** [open-horizon/anax#2100]
Agent-install.sh is failing to upgrade a device after the most recent changes, here is an example of an agent install upgrading from a clean configuration. Running agent-install.sh **twice** will result in a successful upgrade but running only once results in a failure of anax to start.

Please see the attached logs which captures the state of the logs from anax after agent-install.sh runs the first time.

[debug-data.txt](https://github.com/open-horizon/anax/files/5271211/debug-data.txt)
[failure-output-agent-install.txt](https://github.com/open-horizon/anax/files/5271217/failure-output-agent-install.txt)

**Comment by omega2733** [comment]
New PR was submitted, will test with new code and report back if issue persists. If issue is solved will close.

**Comment by omega2733** [comment]
Just tested this with the latest master code (116a6ff) which had some agent-install.sh changes merged since last tested. Testing this code resulted in the same issue during upgrade, however now running agent-install.sh more than once still results in an anax failure, previously this would upgrade the agent with a running client.

The same errors were returned, please see the attached logs for this run.
[output-debug-data-new-code.txt](https://github.com/open-horizon/anax/files/5271714/output-debug-data-new-code.txt)

**Comment by omega2733** [comment]
Requested output of journalctl -xe "_SYSTEMD_UNIT=horizon.service". Appears anax is restarting due to some missing configuration? 

`Terminating, unable to start model management resource manager. Please set either CSSURL in the anax configuration file or HZN_FSS_CSSURL in /etc/default/horizon file`

[debug-data-new-code.txt](https://github.com/open-horizon/anax/files/5271740/debug-data-new-code.txt)

**Comment by bmpotter** [comment]
@omega2733 I think this has been fixed with recent versions of `agent-install.sh`, because it now establishes the new `/etc/default/horizon` settings before starting or restarting the agent. Please retest.

**Comment by omega2733** [comment]
Still have an issue with this with the latest code attempting to upgrade from a lower version to higher version. I had a successfully running env (with agreement), then upgraded to a new env. This time hzn env came back with only a single value set  (anax port) preventing horizon from starting, all other config values that previously existed were now blank.

I was able to workaround this by manually entering the values, restarting horizon which successfully formed an agreement. This shows it was just the missing /etc/default/horizon values preventing startup. Please see the attached log with the failure and /etc/default/horizon output.

[upgrade-failure-blank-config.txt](https://github.com/open-horizon/anax/files/5356040/upgrade-failure-blank-config.txt)

I've also noticed one other new regression, when installing this from a brand new configuration (no currently installed horizon packages). I'll open a new ticket for this and reference here shortly.

**Comment by omega2733** [comment]
Actually instead of opening a new issue, I'm just going to comment here because these two issues after further consideration seem to be related. With a brand new config we have run agent-install.sh twice to successfully get a working /etc/default/horizon config. 

Running it only once results in the below error and a blank config as well. Running it a second time (from a blank config, non upgrade) succeeds. I'd wager that running this in an upgrade configuration would have the same result and appear to be the same cause.

`2020-10-09 08:50:55 Done: Horizon agent ready
2020-10-09 08:50:55 Registering the edge node...
hzn register -m 'jingoes1' -o 'mycluster' -u 'iamapikey:<redacted>' -n 'jingoes1:'    -p 'mycluster/mcgj-test-pattern'
Error: Could not get the Exchange url from environment variable HZN_EXCHANGE_URL or the horizon agent`

**Comment by omega2733** [comment]
Re-tested with a newer package per Bruce (old packages were 2.27.0-93, 2.27.0-94) (new packages are 2.27.0-140, 2.27.0-141) and this works fine with the new packages. The above errors were encountered with the older package versions.

If this is intended behavior let's close the issue but make sure we document. I also need to test an IEAM upgrade from 4.1 -> -141 as I expect we'll see the same issue so we'll need to document this if so.

**Comment by omega2733** [comment]
Tested upgrading from IEAM 4.1 -> 2.27.0-140 and this was successful with no issues so our previous customers shouldn't have any issues upgrading which is great news!
